ajax_with_jquery_exercise/script.js

```diff
@@ -0,0 +1,338 @@
+            } else {
+                $newSpan.attr('class', "glyphicon glyphicon-star-empty");
+            }
+        }
+        
+
+        let $newNumSpan = $("<span>")
+            .addClass("number");
+        if (faveStatus === false){
+            $newNumSpan.text($("#theList>div").length+1 + ". ");
+        };
+
+
+        let $newA = $("<a>", {
+            href: displayUrl, 
+            target: "_blank",
+            text: title 
+        });
+
+        let url = $newA.attr('href');
+
+        if (url !== undefined) {
+            url = url.split("://")[1].split("/")[0];
+            if (url.split(".").length > 2){
+                url = url.split(".").slice(-2).join(".");
+            } 
+        } else {
+            url = "none"
+        }
+        
+
+        let $newUrlSpan = $("<span>")
+            .attr('data-storyId', storyId)
+            .attr('data-by', by)
+            .attr('data-domain', url)
+            .text("(" + url + ")");
+        if(title in favObj) {
+            $newUrlSpan.attr('data-id', favObj[title])
+        }
+        
+        let $newUrlSmall = $("<small>")
+            .append($newUrlSpan);
+
+
+        let $newDiv = $("<div>")
+            .append($newNumSpan)
+            .append($newSpan)
+            .append($newA)
+            .append($newUrlSmall)
+
+        $listSection.append($newDiv);
+    }
+
+
+    // ---- EVENT LISTENERS ----
+
+    // 1. Slide login screen up/down.
+
+    $login.on('click', function(e){
+        // this just rolls out the form.
+
+        if (submitToggle === false){
+            submitToggle = true;
+            $slideScreen.slideDown();
+        } else {
+            submitToggle = false;
+            $slideScreen.slideUp();
+        }
+
+    })
+
+    // 2. Log in an existing user.
+
+    $loginForm.on('submit',function(e){
+
+        e.preventDefault();
+        // create new div. as a new last child to 
+        // send data to heroku app
+        // take token returned and stick into localstorage
+        $.ajax({
+            method: "POST",
+            url: "https://hn-favorites.herokuapp.com/login",
+            data: {email:$('#email').val(),password:$('#password').val()},
+            dataType: "json"
+        }).then(function(res){
+            localStorage.setItem("token", res.auth_token);
+            authToken = res.auth_token;
+            toggleNav();
+            favObj = JSON.parse(localStorage.getItem("favObj"));
+            loadTopStories(); // ----------- need to use local storage to retrieve
+        })
+        .catch(function(err){
+            alert("Error logging in. Please check your credentials and try again.")
+            $loginForm.trigger("reset")
+        });
+
+        $loginForm.trigger("reset");
+
+        submitToggle = false;
+        $slideScreen.slideUp();
+
+    });
+
+    // 3. Let a new user sign up.
+
+    $newUserForm.on('submit',function(e){
+
+        e.preventDefault();
+
+        $.ajax({
+            method: "POST",
```

Comment:
could you wrap this code and the one above in a single function? Lots of duplication going on here